tests/test_nessie_cli.py

```diff
@@ -354,41 +363,24 @@ def test_merge_json() -> None:
     assert_that(merge_response.target_branch).is_equal_to("main")
     assert_that(merge_response.expected_hash).is_equal_to(main_hash)
     assert_that(merge_response.effective_target_hash).is_equal_to(main_hash)
-    assert len(merge_response.source_commits) == 1
-    assert len(merge_response.details) == 1
-    assert_that(merge_response.details[0].source_commits[0]).is_equal_to(dev_hash)
+    assert_that(merge_response.details).is_length(1)
 
     logs = simplejson.loads(execute_cli_command(["--json", "log"]))
     # we don't check for equality of hashes here because a merge
     # produces a different commit hash on the target branch
-    assert len(logs) == 1
-    assert_that(logs[0]["message"]).is_equal_to("test message")
+    assert_that(logs).is_length(2)
+    assert_that(logs[0]["message"]).starts_with("Merged dev at ")
     logs = simplejson.loads(execute_cli_command(["--json", "log", "dev"]))
-    assert len(logs) == 1
-
-
-@pytest.mark.vcr
-def test_merge_legacy() -> None:
-    """Test merge operation."""
-    execute_cli_command(["branch", "dev"])
-    make_commit("merge.foo.bar", _new_table("test_merge"), "dev")
-    main_hash = ref_hash("main")
-
-    merge_output = execute_cli_command(["merge", "dev", "-c", main_hash])
-    assert_that(merge_output).is_equal_to("Merge succeeded but legacy server did not respond with additional details.\n")
```

Comment:
if we remove the test, should we also remove the code around it?
if we keep the code, would it be worth to add a mechanism to run this specific test against an older docker image?